packages/sewing-kit-koa/src/middleware.ts

```diff
@@ -18,22 +18,36 @@ export interface Options {
   serveAssets?: boolean;
   manifestPath?: string;
   caching?: boolean;
+  key?: symbol | string;
 }
 
 const ASSETS = Symbol('assets');
+const DEFAULT_ASSETS = Symbol('default_assets');
 
-export function getAssets(ctx: Context): Assets {
-  return (ctx.state as any)[ASSETS];
+export function getAssets(
+  ctx: Context,
+  key: string | symbol = DEFAULT_ASSETS,
+): Assets {
+  return (ctx.state as any)[ASSETS]?.[key];
```

Comment:
The return type doesn't reflect reality but it was already the case so I'm not sure if we should change the return type 🤔?

Reply:
Isn't it? We're using `setAssets` to set an `Assets` value in there, we're using `getAssets` to retrieve it so it should still be expected to return an `Assets` value. Unless you mean because it's potentially undefined?